### PR TITLE
PADV-528 - Integrate Sidebar MFE in Learning MFE (Migration).

### PR DESCRIPTION
**Pull request by anfbermudezme**
## Ticket

https://agile-jira.pearson.com/browse/PADV-526

## Description

We will deploy the Learning (Courseware) MFE: https://github.com/openedx/frontend-app-learning/tree/open-release/olive.4 for the Olive migration, so it's necessary add the sidebar navigation in the olive.4 release.

## Type of Change

- [x] Cherry pick https://github.com/Pearson-Advance/frontend-app-learning/pull/1
- [x] Cherry pick https://github.com/Pearson-Advance/frontend-app-learning/pull/3

## Testing

- Use `make dev.up `to enable platform containers.
- Sign in in `localhost:18000`
- Go to `http://localhost:2000/course/your_course_key`.
- Test the Outline Sidebar Navigation.

### CODE REVIEW

**Review by alexjmpb — APPROVED** [review]

File: .github/workflows/validate.yml [hunk 1]
```diff
@@ -1,11 +1,9 @@
 name: validate
 on:
   push:
-    branches:
-    - master
+    branches: [pearson-release/olive.master]
```
Comment (Squirrel18) [inline .github/workflows/validate.yml]: We don't have an "olive.master" branch. @anfbermudezme

File: .github/workflows/validate.yml [hunk 2]
```diff
@@ -1,11 +1,9 @@
...
   pull_request:
-    branches:
-    - '**'
+    branches: [pearson-release/olive.master]
```
Comment (Squirrel18) [inline .github/workflows/validate.yml]: We don't have an "olive.master" branch. @anfbermudezme

File: src/courseware/CoursewareContainer.jsx [hunk 3]
```diff
@@ -269,6 +271,13 @@ class CoursewareContainer extends Component {
     if (nextSequence !== null) {
       history.push(`/course/${courseId}/${nextSequence.id}/first`);
 
+      // Function to send the event to Outline Navigation SIdebar.
```
Comment (Squirrel18) [inline src/courseware/CoursewareContainer.jsx]:
```suggestion
      // Function to send the event to Outline Navigation Sidebar.
```

File: src/courseware/CoursewareContainer.jsx [hunk 4]
```diff
@@ -278,11 +287,30 @@ class CoursewareContainer extends Component {
 
   handlePreviousSequenceClick = () => {
     const { previousSequence, courseId } = this.props;
+
     if (previousSequence !== null) {
       history.push(`/course/${courseId}/${previousSequence.id}/last`);
+
+      // Function to send the event to Outline Navigation SIdebar.
```
Comment (Squirrel18) [inline src/courseware/CoursewareContainer.jsx]:
```suggestion
      // Function to send the event to Outline Navigation Sidebar.
```

File: src/courseware/course/sidebar/TriggerButton.jsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import React, { useContext } from 'react';
+import SidebarContext from './SidebarContext';
+import { SIDEBARS } from './sidebars';
```
Comment (Squirrel18) [inline src/courseware/course/sidebar/TriggerButton.jsx]: Why not use absolute imports? @anfbermudezme

File: src/courseware/course/sidebar/sidebars/outline/OutlineSidebar.jsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
+import { ensureConfig, getConfig } from '@edx/frontend-platform';
+import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
+import React, { useContext } from 'react';
+import SidebarBase from 'courseware/course/sidebar/common/SidebarBase';
+import SidebarContext from 'courseware/course/sidebar/SidebarContext';
+import { ID } from 'courseware/course/sidebar/sidebars/outline/OutlineTrigger';
+
+import messages from './messages';
```
Comment (Squirrel18) [inline src/courseware/course/sidebar/sidebars/outline/OutlineSidebar.jsx]: Why not use absolute imports? @anfbermudezme

**Review by Squirrel18 — CHANGES_REQUESTED** [review]

**Review by Squirrel18 — APPROVED** [review]

**Pull request merged**